### PR TITLE
docs: revise 1.0 release notes

### PR DESCRIPTION
**Pull request by majorgreys**
[Preview of revised 1.0 release notes](https://output.circle-artifacts.com/output/job/afd17c78-0275-4ff8-bde2-f6609875763c/artifacts/0/tmp/docs/_release_notes_all.html)

[Preview of 1.0 release notes without duplicate reno notes](https://output.circle-artifacts.com/output/job/83600983-7be0-4614-b521-4ee2cbb94589/artifacts/0/tmp/docs/release_notes.html)

### CODE REVIEW

**Comment by majorgreys** [comment]
@Mergifyio backport 1.0

**Review by brettlangdon — DISMISSED** [review]

File: docs/advanced_usage.rst [hunk 1]
```diff
@@ -306,10 +306,21 @@ Before upgrading, we recommend resolving any deprecation warnings raised in your
...
+
+To identify the deprecated environment variables, search a code base for the following patterns::
+
+    git grep -e "DATADOG_[A-Z_]*" \
+      -e "DATADOG_SERVICE_NAME" \
+      -e "DD_CALL_BASIC_CONFIG" \
+      -e "DD_LOGGING_RATE_LIMIT" \
+      -e "DD_SERVICE_NAME" \
+      -e "DD_TRACER_PARTIAL_FLUSH_ENABLED" \
+      -e "DD_TRACER_PARTIAL_FLUSH_MIN_SPANS"
```
Comment (brettlangdon) [inline docs/advanced_usage.rst:322]: Should this come after the pytest/warnings stuff? or before L312?

Basically it reads as: "You can not enable warnings, here is how to grep for env variable usage"

Comment (majorgreys) [reply docs/advanced_usage.rst]: Fair point

**Review by brettlangdon — APPROVED** [review]

File: docs/advanced_usage.rst [hunk 2]
```diff
@@ -306,22 +306,22 @@ Before upgrading, we recommend resolving any deprecation warnings raised in your
 
 .. _upgrade-0.x:
 
-0.x Release Line
-^^^^^^^^^^^^^^^^
+Upgrading from 0.x to 1.x
+^^^^^^^^^^^^^^^^^^^^^^^^^
 
-As of v0.59.0, you can enable warning filters for ddtrace library deprecations.
+As of v0.60.0, you can use warning filters for ``ddtrace.DDTraceDeprecationWarning`` to update your code.
```
Comment (Kyle-Verhoog) [inline docs/advanced_usage.rst:312]:
```suggestion
As of v0.60.0, you can use warning filters for ``ddtrace.DDTraceDeprecationWarning`` to find and replace usages of deprecated features.
```

wording sounded a little off before

File: docs/advanced_usage.rst [hunk 3]
```diff
@@ -306,22 +306,22 @@ Before upgrading, we recommend resolving any deprecation warnings raised in your
 
 .. _upgrade-0.x:
 
-0.x Release Line
-^^^^^^^^^^^^^^^^
+Upgrading from 0.x to 1.x
+^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (Kyle-Verhoog) [inline docs/advanced_usage.rst:311]: Should we basically say that you need to upgrade to 0.60 as a prerequisite? Most of the following instructions only apply to that version.

Comment (majorgreys) [reply docs/advanced_usage.rst:311]: With enabling all deprecation warnings and grepping for environment variables, upgrading to 0.60 is not a requirement. In the 1.0 release notes we make it more clear our recommendation to upgrade to 0.60. This section is more for guidance on how to migrate, whether you can upgrade to 0.60 or not.

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**